frontend/src/components/BulkActionModel.tsx

```diff
@@ -180,37 +181,44 @@ export function BulkActionModel<T = unknown>(props: IBulkActionModelProps<T> | {
                                   <AcmSubmit
                                       key="submit-bulk-action"
                                       id="submit-button"
-                                      isDisabled={props.confirmText !== undefined && confirm !== props.confirmText}
+                                      isDisabled={
+                                          !props.resources?.length ||
+                                          (props.confirmText !== undefined && confirm !== props.confirmText)
+                                      }
                                       variant={props.isDanger ? ButtonVariant.danger : ButtonVariant.primary}
                                       onClick={async () => {
-                                          setProgressCount(props.resources.length)
-                                          const requestResult = resultsSettled(
-                                              props.resources.map((resource) => {
-                                                  const r = props.actionFn(resource)
-                                                  return {
-                                                      promise: r.promise.finally(() =>
-                                                          setProgress((progress) => progress + 1)
-                                                      ),
-                                                      abort: r.abort,
-                                                  }
-                                              })
-                                          )
-                                          const promiseResults = await requestResult.promise
                                           const errors: ItemError<T>[] = []
-                                          promiseResults.forEach((promiseResult, index) => {
-                                              if (promiseResult.status === 'rejected') {
-                                                  let validError = true
-                                                  if (props.isValidError) {
-                                                      validError = props.isValidError(promiseResult.reason)
-                                                  }
-                                                  if (validError) {
-                                                      errors.push({
-                                                          item: props.resources[index],
-                                                          error: promiseResult.reason,
-                                                      })
+                                          if (props.customFn) {
```

Comment:
I'm not sure that this component should support custom functions; seems like it defeats the purpose of using it in the first place?  Is it possible to just leverage the existing code?

Reply:
removed the customFn, but I still needed a preActionFn to run before looping through the actionFn in order to create the namespace that the bare metal assets use--we could create that namespace in each actionFn but it wouldn't be performant if there are a hundred assets imported that only needed one namespace to be created